src/options.c

```diff
@@ -351,7 +351,8 @@ static void print_usage(char *name)
 
 	printf(JOHN_USAGE, name);
 #if defined(HAVE_OPENCL) || defined(HAVE_CUDA)
-	printf("%s", JOHN_USAGE_GPU);
+	if (!none_opencl_device)
+		printf("%s", JOHN_USAGE_GPU);
```

Comment:
In case someone has CUDA but (for whatever reason) not OpenCL, we need to handle that.